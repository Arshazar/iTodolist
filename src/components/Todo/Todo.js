import React, { useState } from 'react'
import { StyledTodo } from './Todo.styled'
import ShowTodo from '../../containers/ShowTodo'

const Todo = ({ completed, text, id }) => {
    const [modal, setModal] = useState(false)
    const toggle = () => setModal(!modal)

    return (
        <>
            <StyledTodo>
                <li
                    onClick={() => setModal(true)}
                    style={{
                        backgroundColor: completed ? '#86fcbd' : '',
                    }}
                >
                    <p>{text}</p>
                </li>
            </StyledTodo>
            {modal ? (
                <ShowTodo
                    completed={completed}
                    id={id}
                    text={text}
                    modal={modal}
                    toggle={toggle}
                />
            ) : (
                <div />
            )}
        </>
    )
}

export default Todo
