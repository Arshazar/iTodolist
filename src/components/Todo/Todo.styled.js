import styled from 'styled-components'

export const StyledTodo = styled.div`
    li {
        display: flex;
        flex-wrap: row;
        height: 40px;
        border-bottom: solid 1px lightgrey;
        padding: 5px;

        p {
            margin: auto;
            margin-left: 10px;
        }

        &:hover {
            cursor: pointer;
            background: ${({ theme }) => theme.hoverItemColour};
        }
    }
`
