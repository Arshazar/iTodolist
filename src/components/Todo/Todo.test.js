import React from 'react'
import { shallow } from 'enzyme'
import Todo from './Todo'

it('renders todo component', () => {
    expect(shallow(<Todo />)).toMatchSnapshot()
})
