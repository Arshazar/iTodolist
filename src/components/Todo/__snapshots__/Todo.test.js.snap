// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders todo component 1`] = `ShallowWrapper {}`;
